src/Mod/Spreadsheet/TestSpreadsheet.py

```diff
@@ -547,6 +638,29 @@ def test_cath(self):
         self.assertTrue(self.sheet.C27.startswith("ERR: Units must be equal"))
         self.assertMostlyEqual(self.sheet.D27, Units.Quantity("3 mm"))
 
+    def test_not(self):
+        self.sheet.set("A20", "=not(3)")
+        self.sheet.set("B20", "=not(-3)")
+        self.sheet.set("C20", "=not(-3.5)")
+        self.sheet.set("D20", "=not(3mm)")
+        self.sheet.set("E20", "=not(3.5mm)")
+        self.sheet.set("F20", "=not(-3.5mm)")
+        self.sheet.set("G20", "=not(0)")
+        self.sheet.set("H20", "=not(0mm)")
+        self.sheet.set("I20", "=not(1)")
+
+        self.doc.recompute()
+
+        self.assertEqual(self.sheet.A20, 0)
+        self.assertEqual(self.sheet.B20, 0)
+        self.assertEqual(self.sheet.C20, 0)
+        self.assertEqual(self.sheet.D20, 0)
+        self.assertEqual(self.sheet.E20, 0)
+        self.assertEqual(self.sheet.F20, 0)
+        self.assertEqual(self.sheet.G20, 1)
+        self.assertEqual(self.sheet.H20, 1)
+        self.assertEqual(self.sheet.I20, 0)
```

Comment:
We don't really have a boolean type, but we do have dimensionless/unitless numbers. How about only allowing these to be used as input in these functions?
Or is there any benefit of allowing any type of quantity to be used as a boolean?